### ISSUE TITLE
Pub failed to delete entry on clevertap_plugin ^2.4.1

### ISSUE THREAD

**Issue by danimnh** [CleverTap/clevertap-flutter#260]
Hi, I wanted to start using this lib but got some issues on my end when installing dependencies on 
```
dependencies:
   clevertap_plugin: ^2.4.1
```

Here are the error message
```
$ flutter pub get

Downloading packages... (3.3s)
Pub failed to delete entry because it was in use by another process.
This may be caused by a virus scanner or having a file
in the directory open in another application.
```

[similar cases](https://stackoverflow.com/questions/67578189/pub-failed-to-delete-entry-because-it-was-in-use-by-another-process)
What I did try to solve this on Windows
1. `start ms-settings:developers`
2. Run my terminal and/or PowerShell as admin
3. Run vs code as admin

with no good results, but I find out that using `clevertap_plugin: ^1.9.1` works on windows, no idea on iOS though. 

Does anyone happen to know what's the case here and why using specific version works?  Thank you!

<details>
<summary>Flutter Doctor</summary>
<br>

```
$ flutter doctor -v
[✓] Flutter (Channel master, 3.23.0-13.0.pre.290, on Microsoft Windows [Version 10.0.22631.4169], locale en-ID)
    • Flutter version 3.23.0-13.0.pre.290 on channel master at C:\Users\danim\fluttersdk\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c63339866d (3 months ago), 2024-06-20 18:09:08 -0500
    • Engine revision dda82d905f
    • Dart version 3.5.0 (build 3.5.0-284.0.dev)
    • DevTools version 2.36.0

[✓] Windows Version (Installed version of Windows is version 10 or higher)

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at C:\Users\danim\AppData\Local\Android\sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11572160)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.10.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.10.35013.160
    • Windows 10 SDK version 10.0.22621.0

[✓] Android Studio (version 2023.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.10+0--11572160)

[✓] VS Code (version 1.93.1)
    • VS Code at C:\Users\danim\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.98.0

[✓] Connected device (4 available)
    • SM A507FN (mobile) • RR8MA065SWF • android-arm64  • Android 11 (API 30)
    • Windows (desktop)  • windows     • windows-x64    • Microsoft Windows [Version 10.0.22631.4169]
    • Chrome (web)       • chrome      • web-javascript • Google Chrome 129.0.6668.59
    • Edge (web)         • edge        • web-javascript • Microsoft Edge 129.0.2792.65

[✓] Network resources
    • All expected network resources are available.
```
</details>

**Comment by y0av** [comment]
happens to me as well, exactly as @danimnh described.

Flutter (Channel stable, 3.24.3, on Microsoft Windows [Version 10.0.19045.4894])

**Comment by rozabond** [comment]
happens to me as well, exactly as @danimnh described.

Flutter (Channel stable, 3.24.3, on Microsoft Windows [Version 10.0.19045.4894])

**Comment by Anush-Shand** [comment]
@y0av @rozabond Could you also please share the version of `dart `used

**Comment by danimnh** [comment]
@Anush-Shand mine is
```
Dart SDK version: 3.5.0-284.0.dev (dev) (Thu Jun 20 09:06:36 2024 -0700) on "windows_x64"
```

**Comment by rozabond** [comment]
@Anush-Shand 

Flutter 3.24.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 5874a72aa4 (8 weeks ago) • 2024-08-20 16:46:00 -0500
Engine • revision c9b9d5780d
Tools • Dart 3.5.1 • DevTools 2.37.2

**Comment by y0av** [comment]
@Anush-Shand 
Dart SDK version: 3.5.3 (stable) (Wed Sep 11 16:22:47 2024 +0000) on "windows_x64"

**Comment by KambleSonam** [comment]
@danimnh @y0av @rozabond The issue is fixed in flutter version 2.5.0.
Kindly update the dependency to 2.5.0 and confirm.
`clevertap_plugin: 2.5.0`

**Comment by danimnh** [comment]
@KambleSonam I can confirm that `flutter pub get` now works after using `2.5.0`. Thank you for your kind help!
